src/app/reporting/Engine.cpp

```diff
@@ -267,11 +267,11 @@ CHIP_ERROR Engine::BuildSingleReportDataEventReports(ReportDataMessage::Builder
     CHIP_ERROR err    = CHIP_NO_ERROR;
     size_t eventCount = 0;
     TLV::TLVWriter backup;
-    bool eventClean                         = true;
-    ObjectList<EventPathParams> * eventList = apReadHandler->GetEventPathList();
-    EventNumber & eventMin                  = apReadHandler->GetEventMin();
-    EventManagement & eventManager          = EventManagement::GetInstance();
-    bool hasMoreChunks                      = false;
+    bool eventClean                = true;
+    const auto* eventList = apReadHandler->GetEventPathList();
+    EventNumber eventMin                = apReadHandler->GetEventMin();
```

Comment:
This is not right: the code below modifies eventMin, which used to change the state of the ReadHandler, but now will not...

Reply:
Ahhh. Drats.